src/jit/regalloc.cpp

```diff
@@ -141,6470 +141,165 @@ bool Compiler::shouldDoubleAlign(
-                // These integer register(s) immediately die
-                tmpMask = rpPredictRegPick(TYP_INT, PREDICT_REG, lockedRegs | rsvdRegs);
-                if (type == TYP_DOUBLE)
-                {
-                    // For TYP_DOUBLE a second integer register is required
-                    //
-                    tmpMask |= rpPredictRegPick(TYP_INT, PREDICT_REG, lockedRegs | rsvdRegs | tmpMask);
-                }
-
-                // We also need a floating point register that we keep
-                //
-                if (predictReg == PREDICT_NONE)
-                    predictReg = PREDICT_SCRATCH_REG;
-
-                regMask          = rpPredictRegPick(type, predictReg, lockedRegs | rsvdRegs);
-                tree->gtUsedRegs = regMask | tmpMask;
-                goto RETURN_CHECK;
-#endif
-
-            case GT_CNS_INT:
-            case GT_CNS_LNG:
-
-                if (rpHasVarIndexForPredict(predictReg))
-                {
-                    unsigned tgtIndex = rpGetVarIndexForPredict(predictReg);
-                    rpAsgVarNum       = tgtIndex;
-
-                    // We don't need any register as we plan on writing to the rpAsgVarNum register
-                    predictReg = PREDICT_NONE;
-
-                    LclVarDsc* tgtVar   = lvaTable + lvaTrackedToVarNum[tgtIndex];
-                    tgtVar->lvDependReg = true;
-
-                    if (type == TYP_LONG)
-                    {
-                        assert(oper == GT_CNS_LNG);
-
-                        if (tgtVar->lvOtherReg == REG_STK)
-                        {
-                            // Well we do need one register for a partially enregistered
-                            type       = TYP_INT;
-                            predictReg = PREDICT_SCRATCH_REG;
-                        }
-                    }
-                }
-                else
-                {
-#if !CPU_LOAD_STORE_ARCH
-                    /* If the constant is a handle then it will need to have a relocation
-                       applied to it.  It will need to be loaded into a register.
-                       But never throw away an existing hint.
-                       */
-                    if (opts.compReloc && tree->IsCnsIntOrI() && tree->IsIconHandle())
-#endif
-                    {
-                        if (predictReg == PREDICT_NONE)
-                            predictReg = PREDICT_SCRATCH_REG;
-                    }
-                }
-                break;
-
-            case GT_NO_OP:
-                break;
-
-            case GT_CLS_VAR:
-                if ((predictReg == PREDICT_NONE) && (genActualType(type) == TYP_INT) &&
-                    (genTypeSize(type) < sizeof(int)))
-                {
-                    predictReg = PREDICT_SCRATCH_REG;
-                }
-#ifdef _TARGET_ARM_
-                // Unaligned loads/stores for floating point values must first be loaded into integer register(s)
-                //
-      
```

Comment:
Now that this isn't really "shared" other than for the various non-x64/ux targets, this should also be cleaned up in future. There's really no reason that it needs to be special for x64/ux, and there's certainly no need for 3 different methods.